docs/quickstarts/javascript/react/conditional-ui-rendering.mdx

```diff
@@ -0,0 +1,290 @@
+
+Commit, tag and push the changes you made:
+
+```
+git commit -m "updated policy to include visible and enabled clauses"
+git tag v0.0.3
+git push origin master
+git push --tags
+```
+
+## Use The Display State Map middleware
+
+Aserto provides a utility called Display State Map that will return the `visible` and `enabled` decisions from the authorizer to the server and finally to the application. To enable support for the Display State Map, we first need to update the Express.js service: let's add the reference to `displayStateMap` in the object destructuring for the `express-jwt-aserto` dependencies:
+
+Change the line:
+
+```javascript
+const { jwtAuthz } = require("express-jwt-aserto");
+```
+
+to:
+
+```javascript
+const { displayStateMap, jwtAuthz } = require("express-jwt-aserto");
+```
+
+After the `app.use(cors())` statement, add the following line:
+
+```javascript
+app.use(displayStateMap(authzOptions));
+```
+
+## Update the Application
+
+Next, we’ll add the Aserto React SDK to our application:
+
+```
+yarn add @aserto/aserto-react
+```
+
+In `index.js` add the following dependency after the `oidc-react` dependency:
+
+```javascript
+import { AsertoProvider } from "@aserto/aserto-react";
+```
+
+We'll wrap the `<App />` component with the `<AsertoProvider>` provider:
+
+```javascript
+<React.StrictMode>
+  <AuthProvider {...configuration}>
+    <AsertoProvider>
+      <App />
+    </AsertoProvider>
+  </AuthProvider>
+</React.StrictMode>
+```
+
+In `App.js`, we’ll add the `userAserto` hook, as well as a stateful object that will track the result coming back from `getDisplayState`.
+Add the `useAserto` hook dependency below the `useAuth` dependency:
+
+```javascript
+import { useAserto } from "@aserto/aserto-react";
+```
+
+Then, find the code block:
+
+```javascript
+const auth = useAuth();
+const isAuthenticated = auth.userData?.id_token ? true : false;
+const [message, setMessage] = useState(false);
+```
+
+And add the following right after it:
+
+```javascript
+const [displayState, setDisplayState] = useState({
+  visible: false,
+  enabled: false,
+});
+const { init, loading, getDisplayState, error: asertoError } = useAserto();
+```
+
+We’ll add an effect to initialize the Aserto SDK. Add the following after the definition of the `accessSensitiveInformation` callback:
+
+```javascript
+useEffect(() => {
+  async function initAserto() {
+    try {
+      const token = auth.userData?.id_token;
+
+      if (token) {
+        await init({
+          serviceUrl: "http://localhost:8080",
+          accessToken: token,
+          policyRoot: "asertodemo",
+          throwOnError: false,
+        });
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+  if (!asertoError && isAuthenticated) {
+    initAserto();
+  }
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+}, [isAuthenticated, auth.userData?.id_token]);
+```
+
+Next, we’ll add an effect to call `getDisplayState`:
+
+```javascript
```

Comment:
instead of creating this effect, I would just initialize a const displayState when you're sure it's available.